src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -115,11 +115,17 @@ public static double Abs(Complex value)
                 return double.PositiveInfinity;
             }
 
+            return Hypot(value._real, value._imaginary);
+        }
+
+        private static double Hypot (double a, double b)
+        {
             // |value| == sqrt(a^2 + b^2)
```

Comment:
Except when both a and b are infinite: then it returns NaN instead of +infinity because of the inf/inf on line 142. I would move the test for infinity from Abs to Hypot so Abs just calls Hypot and Hypot is correct for all inputs.